src/utils/helpers.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const checkDate = (date) => {
+  const pattern = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/;
+
+  return pattern.test(date);
+};
+
+const validateExpenseBody = (reqBody) => {
+  const { userId, spentAt, title, amount, category, note } = reqBody;
+
+  if (
+    isNaN(userId)
+    || !checkDate(spentAt)
+    || typeof title !== 'string'
+    || isNaN(amount)
+    || typeof category !== 'string'
+    || typeof note !== 'string'
+  ) {
+    return false;
+  }
+
+  return true;
+};
+
+const filterExpenses = (expenses, queries) => {
+  const { from, to, userId, categories } = queries;
+
+  return expenses.filter(expense => {
+    switch (true) {
+      case userId && +expense.userId !== +userId:
```

Comment:
Avoid using unary plus (+) for converting strings to numbers. It may lead to errors in case of invalid inputs. Consider using Number.parseInt or Number.parseFloat for type casting.